src/app/ReadHandler.h

```diff
@@ -130,7 +132,13 @@ class ReadHandler : public Messaging::ExchangeDelegate
     bool IsActiveSubscription() const { return mActiveSubscription; }
     CHIP_ERROR OnSubscribeRequest(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle && aPayload);
     void GetSubscriptionId(uint64_t & aSubscriptionId) { aSubscriptionId = mSubscriptionId; }
-    void SetDirty() { mDirty = true; }
+    AttributePathExpandIterator * GetAttributePathExpandIterator() { return &mAttributePathExpandIterator; }
+    void SetDirty()
+    {
+        mDirty = true;
+        // The interested path in this report is changed, make the path iterator ready to emit paths by reset it.
```

Comment:
```suggestion
        // If the contents of the global dirty set have changed, we need to reset the iterator since the paths
        // we've sent up till now are no longer valid and need to be invalidated.
```